model.py

```diff
@@ -6377,11 +6377,18 @@ def get(cls, _db, url, do_get=None, extra_request_headers=None,
         return representation, False
 
     @classmethod
-    def cacheable_post(cls, _db, url, params, max_age=None):
+    def cacheable_post(cls, _db, url, params, max_age=None,
+                       response_reviewer=None):
         """Transforms cacheable POST request into a Representation"""
 
         def do_post(url, headers, **kwargs):
             kwargs.update({'data' : params})
+            if response_reviewer:
+                # An optional function passed to raise errors if the
+                # post response isn't worth caching.
+                return response_reviewer(
+                    cls.simple_http_post(url, headers, **kwargs)
+                )
```

Comment:
I think response_reviewer is useful enough to go into get(). I think that @aslagle will need something similar to make opds_import.py more reliable--sometimes you try to mirror a URL listed in an OPDS feed and the thing on the other end is just wrong.

It's up to you and @aslagle whether it's better to make this change in this branch or in a future branch.